### ISSUE TITLE
When user types `exit` or press `C-d`, shell close but leave a window behind

### ISSUE THREAD

**Issue by tuhdo** [kyagi/shell-pop-el#27]
To reproduce:

- Pop an eshell window.
- Type `exit`.
- Now we have a useless window.

The way that `multi-term` handles this issue is like this:

```elisp
(defun ansi-term-handle-close ()
  "Close current term buffer when `exit' from term buffer."
  (when (ignore-errors (get-buffer-process (current-buffer)))
    (set-process-sentinel (get-buffer-process (current-buffer))
                          (lambda (proc change)
                            (when (string-match "\\(finished\\|exited\\)" change)
                              (kill-buffer (process-buffer proc))
                              (delete-window))))))
(add-hook 'term-mode-hook 'ansi-term-handle-close)
```

So, we create a process watching for our shell buffer. When it sees `exited` or `finished`, the shell buffer and window are deleted. I think we could use this method for all the shells managed by `shell-pop`.

**Comment by syohex** [comment]
I have merged #28. Please check latest version.

**Comment by tuhdo** [comment]
Excellent. Thank you very much!

**Comment by tuhdo** [comment]
@syohex May you add this feature for `shell`, `ansi-term` and `term` as well, just for consistency? Some people are using those terms.

**Comment by syohex** [comment]
> May you add this feature for shell, ansi-term and term as well, just for consistency? Some people are using those terms.

Yes. I confirmed this feature works for `shell`, `ansi-term`, `term`, `eshell`.

**Comment by tuhdo** [comment]
It does not work for `ansi-term` when the window is maximized. To reproduce:

- Run `shell-pop` with `ansi-term` as default shell.
- `C-x 1` to maximize the shell window.
- Type `exit` or press `C-d` to quit. The shell window remains.

**Comment by syohex** [comment]
@tuhdo I have fixed at #29. Please try latest version.

**Comment by tuhdo** [comment]
Thanks. That fixed it.
